atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraServiceTest.java

```diff
@@ -46,6 +50,22 @@
     private CassandraKeyValueServiceConfig config;
     private Blacklist blacklist;
 
+    @Test
+    public void shouldReturnOnlyLocalHost() {
+        ImmutableSet<InetSocketAddress> hosts = ImmutableSet.of(HOST_1, HOST_2);
```

Comment:
based on the above, I would split this into two tests, one where it will use the default, and one where it will use the new logic. Where the threshold/weighting is 0.0 or 1.0 respectively.

Reply:
It would however be interesting to see our ete docker set up with different dcs and racks, we have a `MultiCassandraTestSuite` but you could easily add to that

and maybe having a test class similar in structure to `CassandraClientPoolIntegrationTest`, where token ranges are exposed and you could actually see pretty quickly how it interacted with different cassandra hosts.

I am also curious, is there a way to tell if a request was serviced "sub-optimally"? i.e. not using the logic you've implemented?

Reply:
I would be interested in a metric of local vs remote calls